packages/runner/injection/cross-origin.js

```diff
@@ -38,17 +40,69 @@ const findCypress = () => {
   }
 }
 
-const Cypress = findCypress()
+// Event listener to echo back the current location of the iframe
+window.addEventListener('message', (event) => {
+  if (event.data === 'aut:cypress:location') {
+    event.ports[0].postMessage(window.location.href)
+  }
+})
+
+// Post the before unload event. We post the load event here instead of in the cross origin communicator
+// because we want to notify the primary cypress instance of unload events even if a corresponding spec bridge
+// has not been created.
+window.addEventListener('beforeunload', () => {
+  parent.postMessage({ event: 'cross:origin:before:unload', data: window.location.origin }, '*')
+})
+
+const handleErrorEvent = (event) => {
+  if (window.Cypress) {
+    // A spec bridge has attached so we don't need to forward errors to top anymore.
+    window.removeEventListener('error', handleErrorEvent)
+  } else {
+    const { error } = event
+    const data = {}
+
+    if (error && error.stack && error.message) {
+      data.message = error.message
+      data.stack = error.stack
+    } else {
+      data.message = error
+    }
+
+    window.top.postMessage({ event: 'cross:origin:aut:throw:error', data }, '*')
```

Comment:
possibly wrap this error with a message about how to catch it (cy.origin prior to visit)